src/deepsparse/loggers_v2/filters/frequency_filter.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from threading import Lock
+from typing import Any, Callable
+
+from .pattern import is_match_found
+
+
+class FrequencyFilter:
+    def __init__(self):
+        self._lock = Lock()
+        self.frequency = {}
+        self.counter = {}
+
+    def add_template_to_frequency(self, tag: str, func: str, rate: int = 1):
+        with self._lock:
```

Comment:
depending on how many logs we end up emitting and at what scale, we might want to find a way to either relax the lock, or have a different lock for keys if this ends up becoming blocking